### DISCUSSION TITLE
Why not base off Far 3 instead of Far 2?

### DISCUSSION THREAD

**Discussion by Rudokhvist** [elfmz/far2l#1890] [Q&A]
Far 3 is a thing for years, and yet for some reason this linux fork still based on Far 2 sources. Maybe it was discussed and answered before, but I was unable to find such topin in discussions. I understand that upgrading from 2 to 3 may require a lot of work, but isn't it worth it considering that Far 3 much more actively developed recently?

**Comment by unxed** [comment]
Historically, far2l was made from the sources of the Far Manager version 2, because the amount of code was half as much of Far 3, and there were no changes significant for users between the second and third versions. At the current moment, you can either port Far 3 to a framework for complex console applications made for far2l, or you can port far2l back to Windows — in both cases you will get a common code base that can be developed further as a single project. But no one user actually needs this; there is no specific user pain that this action would eliminate. If you are interested, take it, we will advise you.

**Reply by Rudokhvist** [reply]
Thank you for explanation.
